grpc/src/main/java/com/linecorp/armeria/internal/client/grpc/ArmeriaClientCall.java

```diff
@@ -478,4 +502,48 @@ private void notifyExecutor() {
             executor.execute(NO_OP);
         }
     }
+
+    private boolean needsDirectInvocation() {
+        return (endpointInitialized || ctx.whenInitialized().isDone()) && ctx.eventLoop().inEventLoop();
+    }
+
+    private void execute(Runnable task) {
+        if (endpointInitialized || ctx.whenInitialized().isDone()) {
+            ctx.eventLoop().execute(task);
+        } else {
+            addPendingTask(task);
+        }
+    }
+
+    private void runPendingTask() {
+        for (;;) {
+            final Runnable pendingTask = this.pendingTask;
+            if (pendingTaskUpdater.compareAndSet(this, pendingTask, NO_OP)) {
+                if (pendingTask != null) {
+                    ctx.eventLoop().execute(pendingTask);
```

Comment:
how about executing right away if `ctx.eventLoop().inEventLoop()` is `true`?

Reply:
If so, the order of execution could be reversed.